src/spec.js

```diff
@@ -252,4 +252,20 @@ describe('lib/readinessManager', () => {
             });
         });
     });
+
+    describe('Chainable', () => {
+        it('Expose public chainable api', () => {
+            const ready = jest.fn();
+            const error = jest.fn();
+
+            readyManager.register('action', () => true);
+
+            const x = readyManager.run()
```

Comment:
why not
`readyManager.register('action', () => true).run().onReady(ready).onError(error)` ?

Reply:
The `register` by design will happen different locations of ur code potentially, where the executor + listeners are expected to run all together